disperser/controller/node_client_manager.go

```diff
@@ -0,0 +1,62 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+type NodeClientManager interface {
+	GetClient(operatorSocket core.OperatorSocket) (clients.NodeClientV2, error)
+}
+
+type nodeClientManager struct {
+	// nodeClients is a cache of node clients keyed by socket address
+	nodeClients *lru.Cache[string, clients.NodeClientV2]
+	logger      logging.Logger
+}
+
+var _ NodeClientManager = (*nodeClientManager)(nil)
+
+func NewNodeClientManager(cacheSize int, logger logging.Logger) (*nodeClientManager, error) {
+	closeClient := func(socket string, value clients.NodeClientV2) {
+		if err := value.Close(); err != nil {
+			logger.Error("failed to close node client", "err", err)
+		}
+	}
+	nodeClients, err := lru.NewWithEvict(cacheSize, closeClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create LRU cache: %w", err)
+	}
+
+	return &nodeClientManager{
+		nodeClients: nodeClients,
+		logger:      logger,
+	}, nil
+}
+
+func (m *nodeClientManager) GetClient(operatorSocket core.OperatorSocket) (clients.NodeClientV2, error) {
+	host, dispersalPort, _, err := core.ParseOperatorSocket(operatorSocket.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse operator socket: %w", err)
+	}
+
+	client, ok := m.nodeClients.Get(dispersalPort)
```

Comment:
Why is it just using the `dispersalPort`? Couldn't this be non-unique?

Reply:
Ah this should just take generic socket